Split jnis into 2 libs and add common lib (#181)
opensearch-project/k-NN: jni/CMakeLists.txt

project(KNNPlugin_JNI)

# ---------------------------------- SETUP ----------------------------------
# Target libraries to be compiled
set(TARGET_LIB_COMMON opensearchknn_common)  # Shared library with common utilities
set(TARGET_LIB_NMSLIB opensearchknn_nmslib)  # nmslib JNI
set(TARGET_LIB_FAISS opensearchknn_faiss)    # faiss JNI
set(TARGET_LIBS "")  # Libs to be installed

set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED True)

option(CONFIG_FAISS "Configure faiss library build when this is on")
option(CONFIG_NMSLIB "Configure nmslib library build when this is on")
option(CONFIG_TEST "Configure tests when this is on")

> if (${CONFIG_FAISS} STREQUAL OFF AND ${CONFIG_NMSLIB} STREQUAL OFF AND ${CONFIG_TEST} STREQUAL OFF)
>     set(CONFIG_ALL ON)
> else()
>     set(CONFIG_ALL OFF)
> endif ()

Reviewer: ```suggestion
if (${CONFIG_FAISS} STREQUAL ON OR ${CONFIG_NMSLIB} STREQUAL ON OR ${CONFIG_TEST} STREQUAL ON)
    set(CONFIG_ALL OFF)
endif ()
```

Author: But then if they are all on, config_all will be set to off.

Reviewer: Good catch!
